modules/dxp/apps/scim/scim-configuration-web-test/src/testIntegration/java/com/liferay/scim/configuration/web/internal/portlet/action/test/SaveScimConfigurationMVCActionCommandTest.java

```diff
@@ -0,0 +1,148 @@
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.scim.rest.util.ScimClientUtil;
+
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Christian Moura
+ */
+@FeatureFlags("LPS-96845")
+@RunWith(Arquillian.class)
+public class SaveScimConfigurationMVCActionCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		ConfigurationTestUtil.deleteConfiguration(_pid);
+	}
+
+	@Test
+	public void testOAuth2TokenIsIssuedToCurrentUser() throws Exception {
+		_company = _companyLocalService.getCompanyById(
+			TestPropsValues.getCompanyId());
+
+		_user = UserTestUtil.addCompanyAdminUser(_company);
+
+		_pid = ConfigurationTestUtil.createFactoryConfiguration(
+			"com.liferay.scim.rest.internal.configuration." +
+				"ScimClientOAuth2ApplicationConfiguration",
+			HashMapDictionaryBuilder.<String, Object>put(
+				"companyId", TestPropsValues.getCompanyId()
+			).put(
+				"matcherField", "email"
+			).put(
+				"oAuth2ApplicationName", "TEST Scim client"
+			).put(
+				"userId", _user.getUserId()
+			).build());
+
+		String scimClientId = ScimClientUtil.generateScimClientId(
+			"TEST Scim client");
+
+		OAuth2Application oAuth2Application =
+			_oAuth2ApplicationLocalService.getOAuth2Application(
+				TestPropsValues.getCompanyId(), scimClientId);
+
+		Assert.assertNotNull(oAuth2Application);
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			new MockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.addParameter(Constants.CMD, "generate");
+
+		mockLiferayPortletActionRequest.addParameter(
+			"oAuth2ApplicationName", "TEST Scim client");
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(_company);
+		themeDisplay.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(_user));
+		themeDisplay.setUser(_user);
+
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		_mvcActionCommand.processAction(
+			mockLiferayPortletActionRequest,
+			new MockLiferayPortletActionResponse());
+
+		List<OAuth2Authorization> oAuth2Authorizations =
+			_oAuth2AuthorizationLocalService.getOAuth2Authorizations(
+				oAuth2Application.getOAuth2ApplicationId(), QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS, null);
+
+		Assert.assertEquals(
+			oAuth2Authorizations.toString(), 1, oAuth2Authorizations.size());
+
+		OAuth2Authorization oAuth2Authorization = oAuth2Authorizations.get(0);
+
+		Assert.assertEquals(_user.getUserId(), oAuth2Authorization.getUserId());
+	}
+
+	@DeleteAfterTestRun
```

Comment:
Looking at the relevant test failures, it seems they are related, because your test is actually deleting the test company:
`com.liferay.portal.kernel.exception.NoSuchCompanyException: No Company exists with the primary key 89940009268660`
[Stacktrace](https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/974160/testReport/com.liferay.scim.rest.resource.v1_0.test/GroupResourceTest/testClientSerDesToDTO/)

No need to make the `_company` variable private, nor delete it afterwards.  As long as we are cleaning up the data we create, we should leave the company intact at the end of the test.